### PR TITLE
[ENG-1449][Hotfix] Update Github client and fix auth depreciation

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Fix our current github addon auth and upgrade our existing library. 

## Changes

- updates githubpy
- fixes and modernizes mocks
- overrides check_authorization method to not use depreciated auth

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify
- Verify

What are the areas of risk?

Any concerns/considerations/questions that development raised?

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-1449

### CODE REVIEW

File: addons/github/api.py [hunk 1]
```diff
@@ -151,7 +151,32 @@ def revoke_token(self):
...
+        auth = (p.get('client_id'), p.get('client_secret'))
+        if self.access_token and auth:
+            url = self.gh3._build_url('user')
+            resp = self.gh3._get(
+                url,
+                headers={
+                    'Authorization': f'token {self.access_token}',
+                    'Accept': 'application/vnd.github.v3+json'
+                },
+                params={'client_id': p.get('client_id'), 'client_secret': p.get('client_secret')},
```
Comment (brianjgeiger) [inline addons/github/api.py:176]: Are these params required for this to work, or does just accessing with the token auth header do all the work here?

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
One question. I'm not super-familiar with the Github v3 auth flow, so I was going through the docs and couldn't find exactly this in there. If it's necessary, cool, but if not, probably best not to pass around those params needlessly.

File: addons/gitlab/tests/test_views.py [hunk 2]
```diff
@@ -6,6 +6,7 @@
 import pytest
```
Comment (mfraezz) [inline addons/gitlab/tests/test_views.py:6]: This is a `gitlab` file, not `github`. It probably doesn't need to be changed.

File: addons/gitlab/tests/utils.py [hunk 3]
```diff
@@ -1,5 +1,8 @@
 import mock
```
Comment (mfraezz) [inline addons/gitlab/tests/utils.py:1]: This is a `gitlab` file, not `github`. It probably doesn't need to be changed.

File: addons/github/api.py (hunk 1 above)
Comment (mfraezz) [reply addons/github/api.py:176]: Appears to only require the `Authorization: token <token>` header

Comment (Johnetordoff) [reply addons/github/api.py:176]: Yep I confirmed it locally

File: addons/gitlab/tests/utils.py (hunk 3 above)
Comment (Johnetordoff) [reply addons/gitlab/tests/utils.py:1]: This is a re-boot of Corbin's old PR and I believe he wrote this to consolidate the mocking, but I can remove it if you want, seems harmless though.

**Review by brianjgeiger — APPROVED** [review]

File: addons/github/api.py [hunk 4]
```diff
@@ -151,7 +151,32 @@ def revoke_token(self):
...
+        False.
+
+        :returns:
+            True if token is valid, False otherwise
+        :rtype:
+            bool
+        """
+        p = self.gh3.session.params
+        auth = (p.get('client_id'), p.get('client_secret'))
+        if self.access_token and auth:
```
Comment (mfraezz) [inline addons/github/api.py:168]: Are `p` or `auth` necessary here? They appear unused. Also:
```
In [1]: bool((None, None))
Out[1]: True
```

Comment (Johnetordoff) [reply addons/github/api.py]: Apparently not, I just had them because they were used in the method this overrides in the client.

**Pull request closed without merging**